File: sources/fsharp/09_fib_recursive/main.fs
// task 09 fib_recursive — expected output: 102334155
// build: dotnet build -c Release    run: dotnet run

let rec fib (n: int64) : int64 =
    if n < 2L then n
    else fib (n - 1L) + fib (n - 2L)

printfn "%d" (fib 40L)

File: sources/csharp/09_fib_recursive/main.cs
// task 09 fib_recursive — expected output: 102334155
// build: coreclr: dotnet build -c Release | nativeaot: dotnet publish -c Release -p:PublishAot=true | mono: mcs -optimize+ main.cs    run: coreclr: dotnet run -c Release (or bin/Release/net8.0/main.exe) | nativeaot: bin/Release/net8.0/publish/main.exe | mono: mono main.exe
using System;

class Program
{
    static long Fib(long n)
    {
        if (n < 2)
        {
            return n;
        }
        return Fib(n - 1) + Fib(n - 2);
    }

    static void Main()
    {
        Console.WriteLine(Fib(40));
    }
}
